### DISCUSSION TITLE
Question: deduplication in blocks inside widget areas?

### DISCUSSION THREAD

**Discussion by dkoo** [Automattic/newspack-blocks#1055] [General]
There is a use case scenario which should be fairly rare but is definitely possible and potentially quite confusing. To illustrate, do the following:

1. Add some Homepage Posts and/or Post Carousel blocks to several widget areas, including the mobile menu, sidebar, and slide-out sidebar.
2. Add several other Homepage Posts and/or Post Carousel blocks to a post or page. Maybe use one of the block patterns which will render many instances and posts. 
3. View the page on the front-end, observe that the posts shown by the blocks in the page's content differ from what's shown in the editor.

The reason this is happening is deduplication: on the front-end, the instances of these blocks inside widget areas are counted when it comes to deduplication behavior. Some of these widget areas (such as the mobile menu and slide-out sidebar) appear ahead of the post content in the markup, so these blocks are rendered first. In the editor, however, the widgets are not rendered, so their posts are not factored into deduplication.

The above is basically working as expected but creates a confusing/alarming scenario where the posts you're seeing in the editor preview could be really different from what you see on the front-end when the widgets are also rendered. How should we handle this? Some options we threw around but didn't decide on fully:

* Disable the use of Homepage Posts and Post Carousel inside certain widget areas (such as mobile menu and slide-out sidebar): not even sure if this is possible
* Disable deduplication behavior in widget block instances: probably not desirable if using the blocks in the regular sidebar, or after-content/footer areas
* Do not handle this scenario at all: the code is working as intended, and it's more of an education issue than a code issue
* Your suggestions here?

**Comment by laurelfulford** [comment]
One possible option 4 could be: try moving the code for the slide-out sidebar and mobile menu to the bottom of the page in the theme, instead of being at the top. It could result in more predictable behaviour (what you see in the editor is what you see on the front-end); though there still might be some unexpected de-duplication on the front end ("I want the latest posts showing up in the slide-out side-bar; why aren't they?") it also could be the lesser of two evils (AKA lesser of two weird de-dupe experiences). 

Edited to add: I still think we need to discourage this overall -- on the off-chance someone is using one of our blocks in the slide-out sidebar area of the theme and including it in the mobile menu, you're still going to get different sets of posts (since on a technical level, it's two blocks, so the same posts can't show in both 😞 ). This suggestion is mostly to improve the WYSIWYG experience for the homepage itself.

**Comment by dkoo** [comment]
Yet another possible option 5 could be to add a new option to Homepage Posts and Post Carousel blocks to toggle on or off deduplication. It would be on by default, so no change in behavior, but if turned off the block will simply run the query according to block attributes while ignoring other block instances on the page, and not be factored into deduplication for other block instances. This might give editors a bit more control over the behavior and would solve for the scenario @laurelfulford outlines above (putting the "same" block in the slide-out sidebar and mobile menu but not being able to show the same posts in both areas).

**Comment by laurelfulford** [comment]
> Yet another possible option 5 could be to add a new option to Homepage Posts and Post Carousel blocks to toggle on or off deduplication.

This has [come up well before there were widget blocks](https://github.com/Automattic/newspack-blocks/issues/258), so it would be helpful beyond that, too!

**Comment by adekbadek** [comment]
Any solution to this might be largely impacted by to change to Full Site Editing.
